### PR TITLE
cli: use ENV variable to load configuration

### PR DESCRIPTION
**Pull request by Exirel**
Remember #1434 I did to work on a new cli interface for Sopel? Well, it gets better and better!

**TL;DR**: I use `SOPEL_CONFIG` env var to load the configuration, using a `sopel.cli.utils` module

## The core of this PR

The core of this PR is to add a `SOPEL_CONFIG` env var to load configuration.

What happens is that:

* first, we try to get the config from the command line argument, like `-c filename`
* then, if the `-c` is not used, we try to look at `SOPEL_CONFIG` env var,
* if nothing is set, we fall back on the `default` config file,

If the `-c` option is used or the `SOPEL_CONFIG` is defined, then it tries to load it, and it won't fall back to the default, ie. if the user decided to use a wrong config file, then Sopel will warn them.

## About the state of the project

To really understand what's going on here, you need to have a look at #1434 because I based this branch from it. Then, here are the good bits of this PR:

* In 614a730 I added `sopel.cli.utils` to store cli-related tools,
* In 3324ca3 I introduce the usage of a `SOPEL_CONFIG` env var,
* In eea0863 I use the 2 previous commit's work to improve the `sopel.run_script` module.

Note: both #1434 and this PR should be rebased after #1429 is merged into master, because the clean-up and the unit-test are very valuable!

**The end goal is to move** both `sopel` and `sopel-module` command line `main` functions into `sopel.cli`.

### CODE REVIEW

**Comment by Exirel** [comment]
For the same reason #1434 is close for now, I close this one. Let's rework things first, then we'll be able to improve the situation here.

**Comment by Exirel** [comment]
Replaced by #1473
